### PR TITLE
add dequeue

### PR DESCRIPTION
**Pull request by RTLer**

### CODE REVIEW

**Comment by amiri27** [comment]
@kamisama when we test on local there's no error.. but when travis test this method it return another assertions...

Could you please help?

**Comment by RTLer** [comment]
Hi,
As you can see on PHP End of Life page PHP 5.3 is already ended and also PHP 5.4 and 5.5 is out of active development.. 

So if it's possible, Could you please update travis-ci to php 5.6 and php7?
Maybe we should keep php 5.5 also..

Regards,

**Comment by kamisama** [comment]
I'll review the PR shortly, it'll take some time, as the code standard change is shadowing the real code change.

File: lib/Resque.php [hunk 1]
```diff
@@ -11,196 +11,364 @@
...
+
+    /**
+     * Pop an item off the end of the specified queues, using blocking list pop,
+     * decode it and return it.
+     *
+     * @param array $queues
+     * @param int $timeout
+     * @return null|array   Decoded item from the queue.
+     */
+    public static function blpop(array $queues, $timeout)
```
Comment (kamisama) [inline lib/Resque.php]: Who is using this new function ?

File: lib/Resque.php [hunk 2]
```diff
@@ -11,196 +11,364 @@
...
+    }
+
+    /**
+     * Remove items of the specified queue
+     *
+     * @param string $queue The name of the queue to fetch an item from.
+     * @param array $items
+     * @return integer number of deleted items
+     */
+    public static function dequeue($queue, $items = [])
```
Comment (kamisama) [inline lib/Resque.php]: I guess that `dequeue($queue)` (without $items) is equivalent to `clear(queue)` ?

File: lib/Resque.php (hunk 1 above)
Comment (amiri27) [inline lib/Resque.php]: Do you mean blpop method?

File: lib/Resque.php (hunk 2 above)
Comment (amiri27) [inline lib/Resque.php]: Yeah, exaclty..

File: lib/Resque.php [hunk 3]
```diff
@@ -11,196 +11,330 @@
...
+     * matching item
+     * item can be ['class'] or ['class' => 'id'] or ['class' => {:foo => 1, :bar => 2}]
+     * @private
+     *
+     * @params string $string redis result in json
+     * @params $items
+     *
+     * @return (bool)
+     */
+    private static function matchItem($string, $items)
```
Comment (kamisama) [inline lib/Resque.php]: What does `$items` exactly represent ? How is it used ?

I got the impression that it is not the same args as the one passed in `push()`.

File: lib/Resque.php [hunk 4]
```diff
@@ -11,196 +11,330 @@
...
+
+        return ($result == 1) ? $counter : 0;
+    }
+
+    /*
+     * Generate an identifier to attach to a job for status tracking.
+     *
+     * @return string
+     */
+    public static function generateJobId()
```
Comment (kamisama) [inline lib/Resque.php]: It seems that function is never used

Comment (kamisama) [inline lib/Resque.php]: I guess we'll keep it, just for php-resque compatibility

**Pull request merged**